onpremise/onpremise.go

```diff
@@ -171,25 +178,102 @@ func WithCustomLogger(logger LogWriter) EngineOptions {
 
 		return nil
 	}
+}
+
+// ToggleFileWatch enables or disables file watching
+// in case 3rd party updates the data file on file system
+// engine will automatically reload the data file
+// default is true
+func ToggleFileWatch(enabled bool) EngineOptions {
+	return func(cfg *Engine) error {
+		cfg.isFileWatcherEnabled = enabled
+		return nil
+	}
+}
+
+// ToggleUpdateOnStart enables or disables update on start
+// default is true
+// if enabled, engine will pull the data file from the distributor on start of the engine
+func ToggleUpdateOnStart(enabled bool) EngineOptions {
+	return func(cfg *Engine) error {
+		cfg.isUpdateOnStartEnabled = enabled
 
+		return nil
+	}
 }
 
-func New(config *dd.ConfigHash, opts ...EngineOptions) (*Engine, error) {
-	manager := dd.NewResourceManager()
+// ToggleAutoUpdate enables or disables auto update
+// default is true
+// if enabled, engine will automatically pull the data file from the distributor
+// if disabled, engine will not pull the data file from the distributor
+// options like WithDataUpdateUrl, SetLicenceKey will be ignored since auto update is disabled
+func ToggleAutoUpdate(enabled bool) EngineOptions {
+	return func(cfg *Engine) error {
+		cfg.isAutoUpdateEnabled = enabled
+
+		return nil
+	}
+}
+
+// ToggleCreateTempDataCopy enables or disables creating a temp copy of the data file
+// default is true
+// if enabled, engine will create a temp copy of the data file and use it to initialize the manager
+// if disabled, engine will use the original data file to initialize the manager
+// this is useful when 3rd party updates the data file on file system
+func ToggleCreateTempDataCopy(enabled bool) EngineOptions {
+	return func(cfg *Engine) error {
+		cfg.isCreateTempDataCopyEnabled = enabled
 
+		return nil
+	}
+}
+
+// SetTempDataDir sets the directory to store the temp data file
+// default is current directory
+func SetTempDataDir(dir string) EngineOptions {
+	return func(cfg *Engine) error {
+		dirFileInfo, err := os.Stat(dir)
+		if err != nil {
+			return fmt.Errorf("failed to get file path: %w", err)
+		}
+
+		if !dirFileInfo.IsDir() {
+			return fmt.Errorf("path is not a directory: %s", dir)
+		}
+
+		cfg.tempDataDir = dir
+		return nil
+	}
+}
+
+// Randomization sets the randomization time in seconds
+// default is 0
```

Comment:
let's set default randomization to 10 minutes, please, as reference implementation specifies that value: https://github.com/51Degrees/pipeline-dotnet/blob/c75b438d0635a63e0fb562e7736659f61741ba73/FiftyOne.Pipeline.Engines/Constants.cs#L85C26-L85C65